README.md

```diff
@@ -73,12 +73,62 @@ const collectionAssetsResponse = await client.listAssets({
 });
 ```
 
+### Generating Stark (Layer 2) keys
+
+Stark keys are required by users in order to transact on ImmutableX's StarkEx Layer 2. They are the equivalent of Ethereum keys on L1, and allows users to sign transactions, and receive and store tokens.
+
+#### Key registration
+
+On ImmutableX, the goal of generating a Stark key is to [register](https://docs.x.immutable.com/docs/how-to-register-users/) a mapping between this key and the user's Ethereum key so that transactions requiring both L1 and L2 signers can be executed by users.
+
+#### How to generate Stark keys on ImmutableX
+
+ImmutableX provides two Stark key generation methods:
+| Type of Stark key generated | How do applications connect to these Stark accounts? | Factors determining when this method should be used: | ImmutableX tools: |
+| --- | --- | --- | --- |
+| [Deterministic](#generating-or-retrieving-a-deterministic-key) - generated using the user's Ethereum key as a seed (which means that the same Ethereum key will always generate the same Stark key) | Users connect with their L1 wallet (ie. Metamask), as the L2 key can simply be obtained from the L1 key. | ***User experience*** - do you want the user to have full responsibility for storing their L2 key?<br/><br/> ***Interoperability*** - how do most applications connect to user's Stark wallets? If a user's key is randomly generated, then they will not be able to connect to an application that requires them to sign in with their L1 key to generate their L2 key.  | [Link SDK](https://docs.x.immutable.com/docs/link-setup)<br/><br/>Core SDK's [`generateLegacyStarkPrivateKey()`](https://github.com/immutable/imx-core-sdk/blob/main/src/utils/stark/starkCurve.ts#L152) method |
+| [Random and non-reproducible](#generating-a-random-non-deterministic-key) - not generated from a user's Ethereum key | Once this Stark key is [registered](#) on ImmutableX (mapped to an Ethereum key), the Stark key owner needs to know and input this. | ***Security*** - because the Stark key generated using this method is completely independent of their Ethereum key, then the compromise of an Ethereum key does not compromise a user's corresponding Stark key.<br/><br/>***Isolated use-case*** - this method is ideal for keys that are only used for one particular functionality, ie. in the backend of an application that allows tokens to be minted from a collection registered with this key. | <br/><br/>Core SDK's [`generateStarkPrivateKey()`](https://github.com/immutable/imx-core-sdk/blob/main/src/utils/stark/starkCurve.ts#L108) method |
```

Comment:
The user is responsible for storing the randomly generated key. If they lose access to the Stark key after randomly generating it, there's no way to retrieve it.

Reply:
This method of generating keys is ideal for wallets that mint assets on the L2 via the backend.